tools/gfx/cuda/render-cuda.cpp

```diff
@@ -0,0 +1,1506 @@
+                break;
+
+            case slang::BindingType::VaryingInput:
+                baseIndex = m_varyingInputCount;
+                m_varyingInputCount += count;
+                break;
+
+            case slang::BindingType::VaryingOutput:
+                baseIndex = m_varyingOutputCount;
+                m_varyingOutputCount += count;
+                break;
+
+            default:
+                baseIndex = m_resourceViewCount;
+                m_resourceViewCount += count;
+                break;
+            }
+
+            BindingRangeInfo bindingRangeInfo;
+            bindingRangeInfo.bindingType = slangBindingType;
+            bindingRangeInfo.count = count;
+            //                bindingRangeInfo.descriptorSetIndex = descriptorSetIndex;
+            //                bindingRangeInfo.rangeIndexInDescriptorSet = slotRangeIndex;
+            //                bindingRangeInfo.subObjectRangeIndex = subObjectRangeIndex;
+            bindingRangeInfo.baseIndex = baseIndex;
+            bindingRangeInfo.descriptorSetIndex = descriptorSetIndex;
+            bindingRangeInfo.rangeIndexInDescriptorSet = rangeIndexInDescriptorSet;
+            bindingRangeInfo.uniformOffset = uniformOffset;
+            m_bindingRanges.add(bindingRangeInfo);
+        }
+
+        SlangInt subObjectRangeCount = typeLayout->getSubObjectRangeCount();
+        for (SlangInt r = 0; r < subObjectRangeCount; ++r)
+        {
+            SlangInt bindingRangeIndex = typeLayout->getSubObjectRangeBindingRangeIndex(r);
+            auto slangBindingType = typeLayout->getBindingRangeType(bindingRangeIndex);
+            slang::TypeLayoutReflection* slangLeafTypeLayout =
+                typeLayout->getBindingRangeLeafTypeLayout(bindingRangeIndex);
+
+            // A sub-object range can either represent a sub-object of a known
+            // type, like a `ConstantBuffer<Foo>` or `ParameterBlock<Foo>`
+            // (in which case we can pre-compute a layout to use, based on
+            // the type `Foo`) *or* it can represent a sub-object of some
+            // existential type (e.g., `IBar`) in which case we cannot
+            // know the appropraite type/layout of sub-object to allocate.
+            //
+            RefPtr<CUDAShaderObjectLayout> subObjectLayout;
+            if (slangBindingType != slang::BindingType::ExistentialValue)
+            {
+                subObjectLayout =
+                    new CUDAShaderObjectLayout(slangLeafTypeLayout->getElementTypeLayout());
+            }
+
+            SubObjectRangeInfo subObjectRange;
+            subObjectRange.bindingRangeIndex = bindingRangeIndex;
+            subObjectRange.layout = subObjectLayout;
+            subObjectRanges.add(subObjectRange);
+        }
+    }
+};
+
+class CUDAProgramLayout : public CUDAShaderObjectLayout
+{
+public:
+    slang::ProgramLayout* programLayout = nullptr;
+    List<RefPtr<CUDAShaderObjectLayout>> entryPointLayouts;
```

Comment:
It might be worth keeping this as a `List` for generality (especially when we add OptiX support into this path), but for the pure CUDA compute case it seems redundant to have multiple entry points here.